submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/stats/web/StatsResource.kt

```diff
@@ -0,0 +1,81 @@
+import org.springframework.web.multipart.MultipartFile
+import uk.ac.ebi.stats.model.SubmissionStat
+import uk.ac.ebi.stats.model.SubmissionStatType
+import uk.ac.ebi.stats.service.SubmissionStatsService
+
+@RestController
+@RequestMapping("/stats")
+@PreAuthorize("isAuthenticated()")
+@Api(tags = ["Submission Stats"])
+class StatsResource(
+    private val statsFileHandler: StatsFileHandler,
+    private val tempFileGenerator: TempFileGenerator,
+    private val submissionStatsService: SubmissionStatsService
+) {
+    @GetMapping("/{type}", produces = [APPLICATION_JSON_VALUE])
+    @ResponseBody
+    @ApiOperation("List all the registered submission stats of the given type")
+    @ApiImplicitParam(name = "X-Session-Token", value = "User authentication token", required = true)
+    fun findByType(
+        @ApiParam(name = "type", value = "The type of stat. It should be one of the following: NUMBER_VIEWS")
+        @PathVariable type: String
+    ): List<SubmissionStat> = submissionStatsService.findByType(SubmissionStatType.valueOf(type))
+
+    @GetMapping("/{type}/{accNo}")
+    @ResponseBody
+    @ApiOperation("List all the registered submission stats of the given type for the given accession")
+    @ApiImplicitParam(name = "X-Session-Token", value = "User authentication token", required = true)
+    fun findByType(
+        @ApiParam(name = "type", value = "The type of stat. It should be one of the following: NUMBER_VIEWS")
+        @PathVariable type: String,
+
+        @ApiParam(name = "accNo", value = "The submission accession number")
+        @PathVariable accNo: String
+    ): SubmissionStat = submissionStatsService.findByAccNoAndType(accNo, SubmissionStatType.valueOf(type))
+
+    @PostMapping
+    @ResponseBody
+    @ApiOperation("Register a stat")
+    @ApiImplicitParam(name = "X-Session-Token", value = "User authentication token", required = true)
+    fun register(
+        @ApiParam(name = "stat", value = "The stat to register")
+        @RequestBody stat: SubmissionStat
+    ): SubmissionStat = submissionStatsService.save(stat)
+
+    @PostMapping("/{type}", headers = ["$CONTENT_TYPE=$MULTIPART_FORM_DATA"])
+    @ResponseBody
+    @ApiOperation("Register the stats contained in the given tab separated file")
+    @ApiImplicitParam(name = "X-Session-Token", value = "User authentication token", required = true)
+    fun register(
+        @ApiParam(name = "type", value = "The type of stat. It should be one of the following: NUMBER_VIEWS")
+        @PathVariable type: String,
+
+        @ApiParam(name = "stats", value = "Tab separated file containing the stats to register")
+        @RequestParam("stats") stats: MultipartFile
+    ): List<SubmissionStat> {
+        val statsFile = tempFileGenerator.asFile(stats)
+        val statsList = statsFileHandler.readStats(statsFile, SubmissionStatType.valueOf(type))
+
+        return statsList.map { submissionStatsService.save(it) }
```

Comment:
I believe service need to support transactional save all stats operations. To guarantee we rolback operation if any fail.

Reply:
Done